src/model/attributes/relations/MorphTo.ts

```diff
@@ -0,0 +1,180 @@
+import { Schema as NormalizrSchema } from 'normalizr'
+import { assert } from '../../../support/Utils'
+import { Schema } from '../../../schema/Schema'
+import { Element, Collection } from '../../../data/Data'
+import { Query } from '../../../query/Query'
+import { Database } from '../../../database/Database'
+import { Model } from '../../Model'
+import { NonEnumerable } from '../../decorators/NonEnumerable'
+import { Relation } from './Relation'
+
+export class MorphTo extends Relation {
+  /**
+   * The database instance.
+   */
+  @NonEnumerable
+  protected _database!: Database
+
+  /**
+   * The field contains all related types.
+   */
+  protected relatedTypes: string[]
+
+  /**
+   * The field name that contains id of the parent model.
+   */
+  protected morphId: string
+
+  /**
+   * The field name that contains type of the parent model.
+   */
+  protected morphType: string
+
+  /**
+   * The associated key of the child model.
+   */
+  protected ownerKey: string
+
+  /**
+   * Create a new morph-to relation instance.
+   */
+  constructor(
+    parent: Model,
+    morphId: string,
+    morphType: string,
+    ownerKey: string
+  ) {
+    super(parent, parent)
+    this.morphId = morphId
+    this.morphType = morphType
+    this.ownerKey = ownerKey
+    this.relatedTypes = []
+  }
+
+  /**
+   * Get all related models for the relationship.
+   */
+  getRelateds(): Model[] {
+    return this.relatedTypes.map((type) => this.$getRelatedModel(type))
+  }
+
+  /**
+   * Define the normalizr schema for the relation.
+   */
+  define(schema: Schema): NormalizrSchema {
+    return schema.union(
+      this.$database().schemas,
+      (_value, parentValue, _key) => {
+        // HACK: Assign missing parent id since the child model is not related back and `attach` will not be called
+        const type: string = parentValue[this.morphType]
+        const model: Model = this.$getRelatedModel(type)
+        const key: string = this.ownerKey || (model.$getKeyName() as string)
+        parentValue[this.morphId] = _value[key as string]
+
+        this.$addNewRelatedType(type)
+
+        return type
```

Comment:
Attaching the parent value to the child here since `attach` will not be correctly called when the parent and child are both set to the same model